### PR TITLE
unreachable! requires a string literal

### PR DESCRIPTION
**Pull request by antiguru**
This restores compatibility with Rust 1.59.0

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - No user-facing changes.

### CODE REVIEW

File: src/repr/src/adt/interval.rs [hunk 1]
```diff
@@ -363,7 +364,8 @@ impl Interval {
                 self.duration -= self.duration % f.nanos_multiplier() as i128;
             }
             Millennium | Century | Decade | Milliseconds | Microseconds => {
-                unreachable!(format!("Cannot truncate interval by {}", f))
+                tracing::error!("Cannot truncate interval by {}", f);
+                unreachable!("Cannot truncate interval")
```
Comment (benesch) [inline src/repr/src/adt/interval.rs:368]: Can you not do:

```
unreachable!("Cannot truncate interval by {f}")
```

or:

```
unreachable!("Cannot truncate interval by {}", f)
```

File: src/repr/src/adt/interval.rs [hunk 2]
```diff
@@ -287,7 +287,8 @@ impl Interval {
             | DateTimeField::Decade
             | DateTimeField::Milliseconds
             | DateTimeField::Microseconds => {
-                unreachable!(format!("Cannot truncate interval by {}", f))
```
Comment (petrosagg) [inline src/repr/src/adt/interval.rs:290]: Shouldn't this become `unreachable!("Cannot truncate interval by {}", f)`?

**Review by benesch — APPROVED** [review]
Thanks! If you do need to stick with `tracing`, can you `use tracing::error` to keep with #10694?

File: src/repr/src/adt/interval.rs (hunk 2 above)
Comment (antiguru) [reply src/repr/src/adt/interval.rs:290]: Yes! Thanks :)

File: src/repr/src/adt/interval.rs (hunk 1 above)
Comment (antiguru) [reply src/repr/src/adt/interval.rs]: Yes, changed it to the first variant.

**Pull request merged**